httphandler/httphandler.go

```diff
@@ -92,17 +87,19 @@ func (h *Handler) validateIncomingRequest(req *http.Request) int {
 // ConfigureHTTPTransport returns http.Transport with customized dialer,
 // MaxIdleConnsPerHost and DisableKeepAlives
 func ConfigureHTTPTransport(conf config.Config) (*http.Transport, error) {
-
-	connDuration, err := time.ParseDuration(conf.ConnectionTimeout)
-	if err != nil {
-		return nil, err
+	maxIdleConnsPerHost := 100
+	if conf.MaxIdleConnsPerHost != 0 {
+		maxIdleConnsPerHost = conf.MaxIdleConnsPerHost
+	}
+	responseHeaderTimeout := 5 * time.Second
```

Comment:
name it defaultResponseHeaderTimeout and move up just below defaultMaxIdleConnsPerHost